pkg/template/source.go

```diff
@@ -87,7 +87,7 @@ func SourcesFromPath(path string, recur bool, encoding Encoding) ([]Source, erro
 		}
 
 		sources[i] = Source{
-			Name:     path,
+			Name:     strings.TrimSuffix(path, filepath.Ext(path)),
```

Comment:
This still doesn't work for something horrendous like `foo.jsonnet.jsonnet`. I have another idea, going to close this and open a new PR.